### PR TITLE
feat(kayenta/google): First pass at support for configuring Google se…

### PR DESCRIPTION
**Pull request by duftler**
…rvice integration.

### CODE REVIEW

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/CommonCanaryGoogleCommandProperties.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1.config.canary.google;
+
+public class CommonCanaryGoogleCommandProperties {
+  public static final String PROJECT_DESCRIPTION = "The Google Cloud Platform project the canary service will use to consume GCS and Stackdriver.";
+
+  public static final String JSON_PATH_DESCRIPTION = "The path to a JSON service account that Spinnaker will use as credentials. "
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/CommonCanaryGoogleCommandProperties.java]: Should we just use the description in the google command properties?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  @Parameter(
+      names = "--stackdriver-enabled",
+      arity = 1,
+      description = "Whether or not to enable Stackdriver as a metrics service (*Default*: `false`)."
+  )
+  private Boolean stackdriverEnabled;
+
+  @Parameter(
+      names = "--metadataCachingIntervalMS",
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java]: Should be kebab-case

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import lombok.AccessLevel;
+import lombok.Getter;
+
+@Parameters(separators = "=")
+public class EditCanaryGoogleCommand extends AbstractConfigCommand {
+  @Getter(AccessLevel.PUBLIC)
+  private String commandName = "edit";
+
+  @Getter(AccessLevel.PUBLIC)
+  private String description = "Edit Spinnaker's canary analysis Google service integration settings.";
```
Comment (lwander) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java]: Do these apply to all canary accounts?

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/canary/CanaryGCSValidator.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      StorageService storageService = new GcsStorageService(
+          n.getBucket(),
+          n.getBucketLocation(),
+          n.getRootFolder(),
+          n.getProject(),
+          jsonPath != null ? jsonPath : "",
+          "halyard",
+          registry);
+
+      System.out.println("*** About to ensure bucket " + n.getBucket() + " exists...");
```
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/canary/CanaryGCSValidator.java]: Intentional?

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/KayentaProfileFactory.java [hunk 5]
```diff
@@ -33,5 +37,55 @@ public SpinnakerArtifact getArtifact() {
...
+      profile.appendContents("kayenta:");
+
+      for (AbstractCanaryServiceIntegration svc : canary.getServiceIntegrations()) {
+        if (svc instanceof GoogleCanaryServiceIntegration) {
+          GoogleCanaryServiceIntegration googleCanaryServiceIntegration = (GoogleCanaryServiceIntegration)svc;
+          boolean isGcsEnabled = googleCanaryServiceIntegration.isGcsEnabled();
+          boolean isStackdriverEnabled = googleCanaryServiceIntegration.isStackdriverEnabled();
+
+          profile
+              .appendContents("  google:")
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/KayentaProfileFactory.java]: Could you use a hashmap to string implementation? There are utils for writing objects as yaml in these profile factories -- check out what gate does

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/CanaryController.java [hunk 6]
```diff
@@ -75,6 +81,8 @@
       @RequestParam(required = false, defaultValue = DefaultControllerValues.validate) boolean validate,
       @RequestParam(required = false, defaultValue = DefaultControllerValues.severity) Severity severity,
       @RequestBody Object rawCanary) {
+    System.out.println("*** rawCanary=" + rawCanary);
```
Comment (lwander) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/CanaryController.java]: Intentional?

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java (hunk 3 above)
Comment (duftler) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java]: The descriptions do, yes. Will pull these out when I add the next service integration.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/CommonCanaryGoogleCommandProperties.java (hunk 1 above)
Comment (duftler) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/CommonCanaryGoogleCommandProperties.java]: Done.

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java (hunk 2 above)
Comment (duftler) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/canary/google/EditCanaryGoogleCommand.java]: Done.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/canary/CanaryGCSValidator.java (hunk 4 above)
Comment (duftler) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/canary/CanaryGCSValidator.java]: Removed.

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/CanaryController.java (hunk 6 above)
Comment (duftler) [reply halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/CanaryController.java]: Removed.

**Comment by duftler** [comment]
@lwander PTAL. Will squash after review.

**Pull request merged**